vcd/resource_vcd_vapp_vm.go

```diff
@@ -257,6 +257,13 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 			Computed:    true,
 			Description: "Operating System type. Possible values can be found in documentation.",
 		},
+		"firmware": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringInSlice([]string{"efi", "bios"}, false),
+			Description:  "Firmware of the VM. Can be either EFI or BIOS, availability depending on the os_type argument. Can be changed only when the VM is turned off",
```

Comment:
It's hard to put in words, as we don't have that written in other fields. I have updated the description, not sure if it's more clear, but it will power off the VM if needed to update.